### ISSUE TITLE
recvAll loops infinitely

### ISSUE THREAD

**Issue by yokodake** [mr/ftp-client#17]
When `retr`ing a file (any file), the program just hangs indefinitely.

The culprit is
https://github.com/mr/ftp-client/blob/9fe0dea886c66373eb13cbbc9d1eb49e7349d50e/ftp-client/src/Network/FTP/Client.hs#L510
I don't know why exactly, but `recv` never hits EOF and still thinks there's data available. this cause recvAll' to loop infinitely by constantly reading an empty bytestring into `chunk`.
I guess an easy fix would be the use hGetContents from Data.ByteString instead, but that'd require changing the `Handle` datatype.
So I don't know what the best fix would be , is what I mean, I wouldn't have minded making a PR otherwise.

**Comment by mr** [comment]
Also I'm curious about what Ftp server you're using and what version

**Comment by yokodake** [comment]
Sorry, I didn't know references in commits ended up on here, I just made that fix so I could continue using the library myself.
The ftp server is unknown, it's from a cloud hosting service, I contacted support and they said they roll their own ftp...

**Comment by mr** [comment]
Hmm, so does it just return empty data indefinitely? I think it would probably be reasonable to terminate when that happens

**Comment by HaskellZhangSong** [comment]
Same problem here
`recvAll` function should be with recursive basecase
```Haskell
recvAll :: (MonadIO m, MonadCatch m) => Handle -> m ByteString
recvAll h = recvAll' ""
    where
        recvAll' bs = ( do
            chunk <- liftIO $ recv h defaultChunkSize
            if chunk /= ""
            then recvAll' $ bs <> chunk
            else return bs)
                `M.catchIOError` (\_ -> return bs)
```

**Comment by mr** [comment]
It looks like it already does this in the conduit version so I think it should be safe. I'll try to get around to this soon.

**Comment by mr** [comment]
Should be fixed in hackage version 0.5.1.3. Let me know if there are any issues.
